### PR TITLE
add cleanup on stack delete for EBS volumes created by persistent storage

### PR DESCRIPTION
**Pull request by jaymccon**
EBS persistent storage module could leave EBS volumes behind if stack was deleted without deleting pv's first. #92

### CODE REVIEW

File: functions/source/CleanupPV/lambda_function.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+def handler(event, context):
+    print('Received event: %s' % json.dumps(event))
+    status = cfnresponse.SUCCESS
+    physical_resource_id = 'PVCleanup'
+    data = {}
+    reason = None
+    try:
+        if event['RequestType'] == 'Delete':
+            print('Removing any orphaned EBS volumes...')
+            tag_name = 'tag:kubernetes.io/cluster/%s' % event['ResourceProperties']['ClusterId']
```
Comment (agamzn) [inline functions/source/CleanupPV/lambda_function.py]: Is this the best tag to be keying off of? It's customer mutable.

File: templates/openshift.template [hunk 2]
```diff
@@ -2290,6 +2300,85 @@
...
+                        }
+                    ]
+                },
+                "ManagedPolicyArns": [
+                    "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+                ],
+                "Path": "/",
+                "Policies": [
+                    {
+                        "PolicyName": "lambda-acm",
```
Comment (agamzn) [inline templates/openshift.template]: ACM? maybe lambda-prune-volumes?

File: functions/source/CleanupPV/lambda_function.py (hunk 1 above)
Comment (agamzn) [reply functions/source/CleanupPV/lambda_function.py]: Correct, that's the concern. I don't see there's a better way either.

**Comment by jaymccon** [comment]
Including this along with additional updates to the 3.9 PR

**Pull request closed without merging**